.devcontainer/Dockerfile

```diff
@@ -0,0 +1,57 @@
+# Use AlmaLinux as the base image
+FROM almalinux:latest
+
+# Update and install necessary packages
+RUN dnf update -y && \
+    dnf install -y \
+    git \
+    python3 \
+    python3-pyyaml \
+    wget \
+    tar \
+    shadow-utils \
+    sudo \
+    && dnf clean all
+
+# Set up arguments for Hugo installation
+ARG HUGO_VERSION=latest
+ARG HUGO_VARIANT=hugo_extended
+ARG WORKSPACE_DIR
+
+# Download and install Hugo
+RUN if [ "$HUGO_VERSION" = "latest" ]; then \
+    HUGO_VERSION=$(curl -s https://api.github.com/repos/gohugoio/hugo/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4)}'); \
+    fi && \
+    ARCH=$(uname -m) && \
+    case "$ARCH" in \
+    "x86_64") HUGO_ARCH="64bit";; \
+    "arm64"|"aarch64") HUGO_ARCH="arm64";; \
+    *) echo "Unsupported architecture: $ARCH" && exit 1;; \
+    esac && \
+    echo "Installing Hugo version: ${HUGO_VERSION}" && \
+    wget -O hugo.tar.gz https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/${HUGO_VARIANT}_${HUGO_VERSION}_Linux-${HUGO_ARCH}.tar.gz && \
+    tar -xzf hugo.tar.gz && \
+    mv hugo /usr/local/bin/hugo && \
+    chmod +x /usr/local/bin/hugo && \
+    rm hugo.tar.gz
+
+# Create a non-root user for development
+ARG USERNAME=vscode
```

Comment:
I'd propose to call the default user "alma" or "hugosite", something more meaningful in my opinion than "vscode"

Reply:
The default base image templates (and most other devcontainer documentation) leverage the `vscode` user by default--so I think anyone familiar with devcontainers wouldn't be confused.

I don't see any reason we cannot switch it--we need to make sure with any future modifications to remember that we're using a "non-standard" username (or just always reference the `UID` I guess?).

Reply:
Ok, I wasn't aware that it is a general "practice" and I'm fine with it

Reply:
I don't know if it's "best practice" as much as "path of least resistance" 😄. 

It would look better with a different username. I just wanted to provide context as to why that was likely chosen :).

Reply:
We can switch it the consensus is to do so for Alma. Like @codyro said, its just a general default on a lot of prefab containers on Docker and various other registries.